### PR TITLE
Upgrade node and node-gyp to build on Windows 11

### PR DESCRIPTION
**Pull request by SpecialAro**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)

5. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Upgrade node and node-gyp to build on Windows 11.

#### Motivation and Context
It was impossible to build Ferdium on Windows 11 without updating node and node-gyp to the latest version.

#### Screenshots
image

#### Checklist

- [x] My pull request is properly named
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ Currently, these are the combinations of system dependencies that work for MacOS
 
 ```bash
 node -v
-v16.14.2
+v18.0.0
```
Comment (vraravam) [inline CONTRIBUTING.md:63]: I was hoping to stick to the latest LTS version of node - which is 16.14.2 - is it possible to try that ? If not, then Im happy to move over to v18 - but, we will then need to change (in a separate PR) all our CI scripts, etc to move to v18

File: package.json [hunk 2]
```diff
@@ -183,7 +183,7 @@
     "jest": "^27.4.7",
     "kebab-case": "^1.0.1",
     "mocha": "^9.1.3",
-    "node-gyp": "8.4.1",
+    "node-gyp": "9.0.0",
```
Comment (vraravam) [inline package.json:186]: Same case here, if we are upgrading to v9 of node-gyp, we will want to also change all our CI scripts as well to reflect the same.

**Comment by SpecialAro** [comment]
I tried with both node and node-gyp at 16.14.2 and 8.4.1 but I always got the same error:
image

Just reverted and tried again and it is the same. I can try on another machine with Windows 11 tomorrow if you think is needed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (vraravam) [reply CONTRIBUTING.md:63]: will resolve based on #19

File: package.json (hunk 2 above)
Comment (vraravam) [reply package.json:186]: will resolve based on #19

**Comment by vraravam** [comment]
thanks for your contribution @SpecialAro! I have also merged the rest of the changes for CI. Please pull the latest and confirm that you can continue building on your windows box.

**Comment by vraravam** [comment]
@all-contributors please add @SpecialAro for build

**Pull request merged**